editor/src/components/canvas/dom-walker.ts

```diff
@@ -94,17 +104,68 @@ function isScene(node: Node): node is HTMLElement {
   )
 }
 
+const LogDomWalkerPerformance = !PRODUCTION_ENV && typeof window.performance.mark === 'function'
+
 export function useDomWalker(props: CanvasContainerProps): React.Ref<HTMLDivElement> {
   const containerRef = React.useRef<HTMLDivElement>(null)
+  const rootMetadataInStateRef = useRefEditorState((store) => store.editor.domMetadataKILLME)
+  const invalidatedTemplatePathsRef = React.useRef<Array<TemplatePath>>([])
   const selectedViews = useEditorState(
     (store) => store.editor.selectedViews,
     'useDomWalker selectedViews',
   )
 
+  const resizeObserver = React.useMemo(() => {
+    return new ResizeObserver((entries: any) => {
+      for (let entry of entries) {
+        const pathAsString = getDOMAttribute(entry.target, UTOPIA_TEMPLATE_PATH)
+        const path = pathAsString != null ? TP.fromString(pathAsString) : null
+        if (
+          path != null &&
+          invalidatedTemplatePathsRef.current != null &&
+          !invalidatedTemplatePathsRef.current.find((invalidPath) =>
+            TP.pathsEqual(path, invalidPath),
+          )
+        ) {
+          invalidatedTemplatePathsRef.current.push(path)
+        }
+      }
+    })
+  }, [])
+
+  const mutationObserver = React.useMemo(() => {
+    return new MutationObserver((mutations: MutationRecord[]) => {
+      for (let mutation of mutations) {
+        if (mutation.attributeName === 'style') {
+          if (mutation.target instanceof HTMLElement) {
+            const pathAsString = getDOMAttribute(mutation.target, UTOPIA_TEMPLATE_PATH)
```

Comment:
on my machine pathAsString is null for elements that have been resized on the canvas. possibly React deleting these mutations from the HTMLElement? 
@enidemi says she saw these template paths there, so maybe there's inconsistent behavior?
this is the key to the changes working.

right now this data-utopia-template-path is appended as a mutation inside the dom walker. however, if we could put them on during canvas render time, it would mean we don't need to worry about loosing these attributes!

@seanparsons I remember we talked about putting the entire template path on the elements, but you said you are dubious if we could ensure that those are correct. do you remember this?